gapic/templates/scripts/fixup_keywords.py.j2

```diff
@@ -21,14 +22,18 @@ def partition(
     return results[1], results[0]
 
 
-class {{ service.client_name }}CallTransformer(cst.CSTTransformer):
+class {{ api.naming.module_name }}CallTransformer(cst.CSTTransformer):
     CTRL_PARAMS: Tuple[str] = ('retry', 'timeout', 'metadata')
-
+    {% with all_methods = [] -%}
+    {% for service in api.services.values() %}{% for method in service.methods.values() -%}
+    {% do all_methods.append(method) -%}
+    {% endfor %}{% endfor -%}
```

Comment:
Is there a reason these lines aren't nested, for example:

```
{% with all_methods = [] -%}
    {% for service in api.services.values() %}
        {% for method in service.methods.values() -%}
            {% do all_methods.append(method) -%}
        {% endfor %}
    {% endfor -%}
    ...
{% endwith %}
```

Reply:
Whitespace control. All of the strangeness in the template is, ultimately, about whitespace control. To the best of my knowledge, [Jinja](https://palletsprojects.com/p/jinja/) is primarily intended to be a template engine for HTML/CSS, which are whitespace insensitive, and which in any case means that the viewer won't see any weird formatting stuff. We have to jump through a few hoops and engage in some tortuous indirection in order to make it generate reasonable, human-viewable Python, and even then the result is sometimes frustrating.